app/jobs/email_privacy_policy_update_job.rb

```diff
@@ -0,0 +1,43 @@
+class EmailPrivacyPolicyUpdateJob < ApplicationJob
+  def perform(petition)
+    terminating = false
+
+    worker = trap("TERM") do
+      terminating = true
+      worker.call
+    end
+
+    Appsignal.without_instrumentation do
+      petition.signatures.find_each do |signature|
+        create_privacy_notification!(signature)
+        send_email(signature)
+
+        if terminating
+          reschedule_job(petition)
+          return true
+        end
+      rescue ActiveRecord::RecordNotUnique
+        next
+      end
+    end
+  ensure
+    trap("TERM", worker)
+  end
+
+  private
+
+  def create_privacy_notification!(signature)
+    PrivacyNotification.create!(id: signature.uuid)
```

Comment:
So we now have the `canonical_email` column which has an email address that's stripped of things like `+` extensions and domain aliases like gmail.com and googlemail.com - not sure whether it's better to try to use a uuid based on that or the standard one based on the entered email. I think on balance the standard one and if people have been using `+` to differentiate their emails by petition that's their problem.